src/jdk.jpackage/share/classes/jdk/jpackage/internal/Arguments.java

```diff
@@ -507,15 +504,6 @@ public boolean processArguments() {
                 }
             }
 
-            if (hasMainJar && !hasMainClass) {
-                // try to get main-class from manifest
-                String mainClass = getMainClassFromManifest();
-                if (mainClass != null) {
-                    CLIOptions.setOptionValue(
-                            CLIOptions.APPCLASS.getId(), mainClass);
-                }
-            }
-
```

Comment:
This duplicates functionality from https://github.com/openjdk/jdk/pull/13413/files#diff-50e3f4f576df4f85d94258043607539f7ec5d55705e7092c00f47432309747a6R210-R218